avocado/core/nrunner.py

```diff
@@ -360,8 +360,53 @@ class BaseRunner(abc.ABC):
     Base interface for a Runner
     """
 
+    @staticmethod
+    def _get_avocado_version():
+        """Return the Avocado package version, if installed"""
+        version = "unknown.unknown"
+        if PKG_RESOURCES_AVAILABLE:
+            try:
+                version = pkg_resources.get_distribution(
+                    'avocado-framework').version
+            except pkg_resources.DistributionNotFound:
+                pass
+        return version
+
+    def set_env_variables(self):
+        """Set default AVOCADO_TEST_* environment variables
+
+        This feature was carried from the legacy runner architecture for
+        compatibility purpose.
+
+        NOTE: these environment variables are redefined inside an Avocado
+        instrumented test.
+        """
+        outputdir = self.runnable.kwargs.get('AVOCADO_TEST_OUTPUTDIR')
+        # create work dir inside the output dir
+        workdir = os.path.join(outputdir, 'workdir')
+        os.mkdir(workdir)
+        # create fake log dir inside output dir
+        logdir = os.path.join(outputdir, 'job-results')
+        os.mkdir(logdir)
+        # create fake log file inside the logdir
+        logfile = os.path.join(logdir, 'debug.log')
+        open(logfile, 'w').close()
+        # add the environment variable values to the runnable kwargs
+        self.runnable.kwargs.update({
```

Comment:
AFAICT we don't have an universal understanding that `kwargs` are intended for environment variables.  This is true for the `exec-test` runner, and possibly others derived from it.  But, for instance, it's not documented or true for `avocado-instrumented`.

Have you thought about that?

Reply:
Yes, I did. As those are particular variable names, I thought it would not hurt. 

The other option I thought about was moving the variables' setup to the `exec` runner. That way, we keep the current behavior of having the environment variables available just for the `exec-test`,  as the `instrumented` tests are already "supported." This is my personal preference, but as I don't see direct damage to creating it in the base runner, I went with that solution.

If that is okay, I'll move the setup to the `exec` runner.

Reply:
> Yes, I did. As those are particular variable names, I thought it would not hurt.
> 
> The other option I thought about was moving the variables' setup to the `exec` runner. That way, we keep the current behavior of having the environment variables available just for the `exec-test`, as the `instrumented` tests are already "supported." This is my personal preference, but as I don't see direct damage to creating it in the base runner, I went with that solution.
> 
> If that is okay, I'll move the setup to the `exec` runner.

Yes, I think moving it to the `exec-test` (and similar? tap?) runners are the way to go at this point.